src/main/java/nl/tudelft/lifetiles/graph/models/sequence/SequenceSegment.java

```diff
@@ -154,6 +160,110 @@ public final long distanceTo(final SequenceSegment other) {
      */
     @Override
     public final int compareTo(final SequenceSegment other) {
-        return Long.compare(this.getStart(), other.getStart());
+        int candidateComp = Long.compare(this.getAbsStart(),
```

Comment:
You won't see the difference in speed if you'd pre-compute them. Don't obfuscate the code for extremely tiny optimizations like these. I agree with you however on the last part, that would increase the time-complexity if pre-computed. I suggest you do make a while loop for the first 5 statements, and keep the `if` for the last one.